src/zarr/core/array.py

```diff
@@ -634,6 +642,14 @@ async def _create_v2(
         if dimension_separator is None:
             dimension_separator = "."
 
+        dtype = parse_dtype(dtype, 2)
```

Comment:
```suggestion
        dtype = parse_dtype(dtype, zarr_format=2)
```
It wasn't clear to me what 2 was doing without the argument name